pydantic/schema.py

```diff
@@ -587,7 +589,9 @@ def field_singleton_sub_fields_schema(
     (EmailStr, {'type': 'string', 'format': 'email'}),
     (UrlStr, {'type': 'string', 'format': 'uri'}),
     (DSN, {'type': 'string', 'format': 'dsn'}),
+    (SecretStr, {'type': 'string', 'format': 'secret-str'}),
     (str, {'type': 'string'}),
+    (SecretBytes, {'type': 'bytes', 'format': 'secret-bytes'}),
```

Comment:
I'm Not sure whether the `format` here has to be one of a specific set of values. if so is `secret-bytes` an allowed value?

Maybe @tiangolo knows?

Reply:
I wasn't sure about this either.

Reply:
I used `name-email` as a reference when I did this, I was not sure about the purpose of these values to be honest, but I included it for the sake of consistency and completeness.

Reply:
## TL;DR:

I would suggest having both as `{"type": "string", "writeOnly": true}`.

---

Sorry for the delay, I hadn't seen the notification. `type` would have to be a valid JSON type: `"null", "boolean", "object", "array", "number", or "string"`.

So, in this case the closest thing would be `"string"`.

`format` is to add more "semantic validation", for constraint information that wouldn't be easily described with normal JSON Schema, like `date`.

The spec allows to add custom format attributes. So, adding custom things in `format` is accepted, although wouldn't convey any special meaning.

But there's an additonal property `writeOnly` made for password fields, I think this would be the most appropriate here.

Reply:
Here's the reference in case you want to check: http://json-schema.org/latest/json-schema-validation.html#rfc.section.10.3

Reply:
I made the changes as needed. @samuelcolvin @tiangolo